### ISSUE TITLE
Why not wrap everything in a container for responsive?

### ISSUE THREAD

**Issue by Jakobud** [adamcoulombe/jquery.customSelect#102]
Instead of creating your `span` next to the `select` like this:

```
<select class='hasCustomSelect'..../>
<span class='customSelect'>...</span>
```

Why don't you wrap everything into a containing element

```
<span class='customSelectContainer'>
  <select class='hasCustomSelect'..../>
  <span class='customSelect'>...</span>
</span>
```

And then make the `.customSelect` and `select` `width: 100%`? That way the user can easily just make the container responsive in whatever way they want and both the select and the customSelect will stay exactly the same size as the container using simple css?

Honestly its kind of weird you put it parallel to the select in the DOM and then you try to manage the width of each element by itself using javascript...

**Comment by deansimcox** [comment]
I agree with you there.

You could do this yourself when initializing though?
`$('.selector').wrap('<div class="customSelectContainer"/>').customSelect();`

**Comment by Jakobud** [comment]
Yeah I'm doing something like that already. I was just curious as to why the plugin wouldn't do something so obvious instead of using JS to hardcode widths.

**Comment by massic80** [comment]
I also thought the same, but implementing the modification would create
compatibility issues with previous installations (and CSS).
I think the solution would be a switch parameter for activating this mode
Massi

2015-09-17 3:33 GMT+02:00 Jake Wilson notifications@github.com:
